app/code/Magento/Store/Console/Command/WebsiteListCommand.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace Magento\Store\Console\Command;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Command\Command;
+
+/**
+ * Class WebsiteListCommand
+ *
+ * Command for listing the configured websites
+ */
+class WebsiteListCommand extends Command
+{
+    /** @var \Magento\Store\Api\WebsiteManagementInterface $storeManager */
+    private $manager;
+
+    public function __construct(
+        \Magento\Store\Api\WebsiteRepositoryInterface $websiteManagement
+    ) {
+        $this->manager = $websiteManagement;
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName('store:website:list')
+            ->setDescription('Displays the list of websites');
+
+        parent::configure();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $table = $this->getHelperSet()->get('table');
+            $table->setHeaders(['ID', 'Default Group Id', 'Name', 'Code', 'Sort Order', 'Is Default']);
+
+            foreach ($this->manager->getList() as $website) {
+                $table->addRow([
+                    $website->getId(),
+                    $website->getDefaultGroupId(),
+                    $website->getName(),
+                    $website->getCode(),
+                    $website->getData('sort_order'),
+                    $website->getData('is_default'),
+                ]);
+            }
+
+            $table->render($output);
+
+            return \Magento\Framework\Console\Cli::RETURN_SUCCESS;
+        } catch (\Exception $e) {
+            $output->writeln('<error>' . $e->getMessage() . '</error>');
```

Comment:
Similar catch logic is already implemented in the CLI entry point script in bin/magento. I would suggest not to duplicate similar code here. The one difference which I see is that this code will output exception message regardless of verbosity. Can this be done using exact exception types for the cases where you want to output exceptions?

One problem which I see is that the return code is just returned from the execute method. CLI framework will consider that this is valid result of the command and will end process successfully.